File: Problem37/euler.py
def is_prime(n):
    if n == 2 or n == 3: return True
    if n < 2 or n%2 == 0: return False
    if n < 9: return True
    if n%3 == 0: return False
    r = int(n**0.5)
    f = 5
    while f <= r:
        if n%f == 0: return False
        if n%(f+2) == 0: return False
        f +=6
    return True 

def lefttoright(n):
    flag = 0
    temp = n
    #print n
    for i in range(len(str(n))):
        temp = temp/10
        #print temp
        if(is_prime(temp)):
            flag = flag +1
    if flag == len(str(n)) - 1:
        return True
    return False

def righttoleft(n):
    temp1 = int(str(n)[::-1])
    flag = 0
    #print n
    for i in range(len(str(n))):
        temp1 = temp1/10
        temp2 =  int(str(temp1)[::-1])
        #print temp
        if(is_prime(temp2)):
            flag = flag +1
    if flag == len(str(n)) - 1:
        return True
    return False

sum = 0
n =  int(raw_input())
for i in range(10, n, 1):
    if(is_prime(i)):
        if lefttoright(i) and righttoleft(i):
            sum = sum + i
print sum
            

File: Problem37/euler.cpp
#include <cmath>
#include <cstdio>
#include <vector>
#include <iostream>
#include <algorithm>
using namespace std;

int prime(long long n)
{
    if(n<2)
        return 0;
    for(int i=2;i<=sqrt(n);i++)
    {
        if(n%i==0)
            return 0;
    }
    return 1;
}

int noofdigits(long long n)
{
    int count=0;
    while(n!=0)
    {
        count++;
        n=n/10;
    }
    return count;
}

int reverse_number(long long n) 
{
    long long reverse = 0;
    while(n != 0) {
        int remainder = n%10;
        reverse = reverse*10 + remainder;
        n/=10;
    }
   return reverse;
}

int main() 
{
    long long n,i,p,sum=0, temp1, temp2, reversed_number;
    int k,d,j;
    cin>>n;
    for(i=11;i<n;i++)
    {
        k=0;
        if(prime(i))
        {
            
            reversed_number = reverse_number(i);
            temp1 = i;
            temp2 = reversed_number;
            k=1;
            d=noofdigits(i);
            for(j=1;j<d;j++)
            {
                if(  !(prime(temp1/10) &&  prime(reverse_number(temp2/10))) )
                    k=0;
                temp1 = temp1/10;
                temp2 = temp2/10;
                
            }
        }
        if(k==1)
        {
            sum = sum + i;
        }
            
    }
    cout<<sum<<endl;
    return 0;
}

